tests/test_book_routes.py

```diff
@@ -28,6 +92,43 @@ def test_create_one_book(client):
     })
     response_body = response.get_json()
 
+    # Assert
+    assert response.status_code == 201
+    assert response_body == {
+        "id": 1,
+        "title": "New Book",
+        "description": "The Best!"
+    }
+
+def test_create_one_book_no_title(client):
+    # Arrange
+    test_data = {"description": "The Best!"}
+
+    # Act & Assert
+    with pytest.raises(KeyError, match="title"):
+        response = client.post("/books", json=test_data)
```

Comment:
Aside from potentially moving these tests to 07b, where they are re-written to check for our custom errors, this really surprised me. I would have expected the response to just give backthe typical 500 internal server error (like we see through postman) rather than exposing the KeyError. Is this behavior just occuring due to using the flask test client?

Reply:
We do get a 500 internal server error when hitting the endpoint in Postman. If we are in production mode Flask hides the KeyError, but in debug mode, it sends the KeyError and stack trace to Postman. 

I believe this is the test client, the test exits as soon as the code crashes and the KeyError gets raised. If we instead tried to write something like:
```python
test_data = {"title": "New Book"}
response = client.post("/books", json=test_data)
assert response.status_code == 500  
```
The test fails as soon as we hit the KeyError and we never get to the assert.

Reply:
That's why this feels leaky to me. Like it feels very strange to write a test that's verifying debug behavior. Essentially, I'm advocating for a way we can avoid showing this, such as moving the failure cases to the next branch where we have better error handling.

Reply:
This may feel "leaky" but it is how Flask is meant to work. I dug into the app and testing config, and this is caused by us setting `"TESTING": True`: https://flask.palletsprojects.com/en/3.0.x/config/#TESTING. If we don't set the testing flag, then we could write a test that checks for a generic 500 error response.

To me, this makes sense for testing; if I'm working on my app, getting a generic 500 error from this test doesn't help me debug an issue nearly as much as knowing there's an error being raised.

Reply:
That's true. It's useful to see the underlying error if it occurs while running a test (now that I think of it, I do see this behavior while working), but I think that's more because it shows us there's a place where an exception _unexpectedly_ escaped. In that case, I would go in and write code to handle the exception (or otherwise figure out why the exception happened) rather than adding a test to receive that exception.

Reply:
That is fair that we'd want to go in an fix that rather than just testing for it. While we could talk around that in class and say we'll address it soon, swapping the lesson order very neatly removes the need for that and any confusion it could cause.

Reply:
I agree. I think you identified the better solution.